File: app/src/app/features/clients/resources/client.resource.ts
import { TrackableResource } from 'src/app/core/resources/trackable.resource';
import { AppointmentResource } from '../../home/resources/appointment.resource';
import { Gender } from '../enums/gender.enum';
import { ClientDiseaseResource } from './client-disease.resource';
import { ClientDueResource } from './client-due.resource';
import { ClientMedicationResource } from './client-mdedcation.resource';
import { ClientObservationResource } from './client-observation.resource';
import { ClientPaymentResource } from './client-payment.resource';
import { ClientSubscriptionsResource } from './client-subscriptions.resource';
import { ClientTagResource } from './client-tag.resource';
import { MeasurementResource } from './measurement.resource';

export class ClientResource extends TrackableResource {
  clientId: string;
  fullName: string;
  birthDate: Date;
  gender: Gender;
  phoneNumber: string;
  isActive: boolean;
  workPLacesIds: Array<number>;
  clientTags: Array<ClientTagResource>;
  clientDiseases: Array<ClientDiseaseResource>;
  clientMedications: Array<ClientMedicationResource>;
  clientObservations: Array<ClientObservationResource>;
  clientMeasurements: Array<MeasurementResource>
  clientSubscriptions: Array<ClientSubscriptionsResource>
  templateId: number;
  payments: Array<ClientPaymentResource>;
  dues: Array<ClientDueResource>
  inDebt: boolean;
  appointments: Array<AppointmentResource>
}

File: api/BeHealth.Business/Resources/Clinets/ClientResource.cs
using BeHealth.Business.Resources.Appointment;
using BeHealth.Presentence.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace BeHealth.Business.Resources.Clinets
{
    public class ClientResource : TrackableResource
    {
        public string ClientId { get; set; }

        public string FullName { get; set; }

        public DateTime? BirthDate { get; set; }

        public GenderEnum? Gender { get; set; }

        public string PhoneNumber { get; set; }

        public bool IsActive { get; set; }

        public List<long> WorkPLacesIds { get; set; }
        public List<TagResource> ClientTags { get; set; }
        public List<ClientDiseaseResource> ClientDiseases { get; set; }
        public List<ClientMedicationResource> ClientMedications { get; set; }
        public List<ClientObservationResource> ClientObservations { get; set; }
        public List<clientMeasurementResource> clientMeasurements { get; set; }
        public List<ClientSubscriptionsResource> ClientSubscriptions { get; set; }
        public List<ClientDueResource> Dues { get; set; }
        public List<ClientPaymentResource> Payments { get; set; }
        public List<AppointmentResource> Appointments { get; set; }

        public long? TemplateId { get; set; }

        public bool InDebt { get; set; }

    }
}
